core/net/rpl/rpl-icmp6.c

```diff
@@ -147,6 +196,30 @@ set16(uint8_t *buffer, int pos, uint16_t value)
   buffer[pos++] = value & 0xff;
 }
 /*---------------------------------------------------------------------------*/
+uip_ds6_nbr_t *
+rpl_icmp6_update_nbr_table(uip_ipaddr_t *from, nbr_table_reason_t reason, void *data)
+{
+  uip_ds6_nbr_t *nbr;
+
+  if((nbr = uip_ds6_nbr_lookup(from)) == NULL) {
+    if((nbr = uip_ds6_nbr_add(from, (uip_lladdr_t *)
+                              packetbuf_addr(PACKETBUF_ADDR_SENDER),
+                              0, NBR_REACHABLE, reason, data)) != NULL) {
+      PRINTF("RPL: Neighbor added to neighbor cache ");
+      PRINT6ADDR(from);
+      PRINTF(", ");
+      PRINTLLADDR((uip_lladdr_t *)packetbuf_addr(PACKETBUF_ADDR_SENDER));
+      PRINTF("\n");
+    }
+  }
+
+  if(nbr != NULL) {
+    /* set reachable timer if we added or found the nbr entry */
+    stimer_set(&nbr->reachable, UIP_ND6_REACHABLE_TIME / 1000);
```

Comment:
Not sure why we want to do this independent of the `reason`. What is the exact semantics of `rpl_icmp6_update_nbr_table`? Do we have a guarantee that, wherever it is called from, we know the neighbor is IPv6-reachable?